pkg/api/vlabs/types.go

```diff
@@ -791,6 +799,16 @@ func (a *AgentPoolProfile) IsUbuntu() bool {
 	return a.IsUbuntu1604() || a.IsUbuntu1804()
 }
 
+// GetAgentPoolByName returns the pool in the AgentPoolProfiles array that matches a name, nil if no match
+func (p *Properties) GetAgentPoolByName(name string) *AgentPoolProfile {
```

Comment:
this function is misplaced, it should be next to `GetAgentPoolIndexByName` and the other functions on `Properties` in the file